proxy_worker/dispatcher.py

```diff
@@ -0,0 +1,528 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+import asyncio
+import concurrent.futures
+import logging
+import os
+import queue
+import sys
+import threading
+import traceback
+import typing
+from asyncio import AbstractEventLoop
+from dataclasses import dataclass
+from typing import Any, Optional
+
+import grpc
+
+from proxy_worker import protos
+from proxy_worker.logging import (
+    CONSOLE_LOG_PREFIX,
+    disable_console_logging,
+    enable_console_logging,
+    error_logger,
+    is_system_log_category,
+    logger,
+)
+from proxy_worker.utils.common import get_app_setting
+from proxy_worker.utils.common import is_envvar_true, get_script_file_name
+from proxy_worker.utils.constants import PYTHON_ENABLE_DEBUG_LOGGING, \
+    PYTHON_THREADPOOL_THREAD_COUNT
+from proxy_worker.version import VERSION
+from .utils.dependency import DependencyManager
+
+# Library worker import reloaded in init and reload request
+_library_worker = None
+
+
+class ContextEnabledTask(asyncio.Task):
+    AZURE_INVOCATION_ID = '__azure_function_invocation_id__'
+
+    def __init__(self, coro, loop, context=None):
+        super().__init__(coro, loop=loop, context=context)
+
+        current_task = asyncio.current_task(loop)
+        if current_task is not None:
+            invocation_id = getattr(
+                current_task, self.AZURE_INVOCATION_ID, None)
+            if invocation_id is not None:
+                self.set_azure_invocation_id(invocation_id)
+
+    def set_azure_invocation_id(self, invocation_id: str) -> None:
+        setattr(self, self.AZURE_INVOCATION_ID, invocation_id)
+
+
+_invocation_id_local = threading.local()
```

Comment:
[Checking] Why local? if application scale to different threadpool, will local storage be an issue and lead to thread safety?

Reply:
This has not related to storage. Its a  variable used to store the data of the current thread